hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPartitionStatsIndex.scala

```diff
@@ -200,6 +199,7 @@ class TestPartitionStatsIndex extends PartitionStatsIndexTestBase {
 
     assertTrue(f1.value.get.get || f2.value.get.get)
     executor.shutdownNow()
+    Thread.sleep(2000L)
```

Comment:
Could we add deterministic conditions?  Why did it fail before?

Reply:
Removed since we have found the real issue.